doc/tags.md

```diff
@@ -64,9 +87,67 @@ RuntimeTagExtension.excluded += MyOtherTag
 
 Everything else will work like the described above.
 
+#### Tag Expressions
+
+Operators (in descending order of precedence)
+
+| Operator | Description | Example |
+|----------|-------------|---------------|
+|!|not|!macos|
+|&|and|linux & integration|
+|&#124;|or|windows &#124; microservice|
+
+
+#### Marking Specs
+
+You can mark all tests in a spec using the tags function in the spec itself.
+
+```kotlin
+class MyTestClass : FunSpec({
+
+  tags(Linux, Mysql)
+
+  test("my test") { } // automatically marked with the above tags
+})
+```
+
+Note: When tagging tests in this way, the Spec will still need to be instantiated in order to retrieve
+the tags. If no root tests are active at runtime, the [beforeSpec](listeners.md) and [afterSpec](listeners.md)
+callbacks will _not_ be invoked.
+
+If you wish to avoid creating the spec class at all then you can annotate a spec using `@Tags(tag1, ...)`.
+Any tags added using this annotation apply to all tests in the class, however this will not stop a class from
+being instantiated unless explicitly excluded.
+
+Consider the following example:
+
+```kotlin
+@Tags(Linux)
+class MyTestClass : FunSpec({
+
+  tags(UnitTest)
+
+  beforeSpec { println("Before") }
+
+  test("A").config(tags = setOf(Mysql)) {}
+  test("B") {}
+})
+```
+
+| Property | Spec Created | Callbacks | Outcome |
+|----------|--------------|---------|----------|
+| kotest.tags=Linux                  | yes | yes | A and B are executed because all tests inherit the `Linux` tag from the annotation |
+| kotest.tags=Linux & Mysql          | yes | yes | A is executed only because all tests have the `Linux` tag, but only A has the `Mysql` tag |
+| kotest.tags=!Linux                 | no  | no  | No tests are executed, and the MyTestClass is not instantiated because we can exclude it based on the tags annotation |
+| kotest.tags=!UnitTest              | yes | no  | No tests are executed because all tests inherit `UnitTest` from the tags function. MyTestClass is instantiated in order to retrieve the tags defined in the class. The beforeSpec callback is not executed because there are no active tests. |
+| kotest.tags=Mysql                  | yes | yes | A is executed only, because that is the only test marked with `Mysql` |
+| kotest.tags=!Mysql                 | yes | yes | B is executed only, because that is the only test marked with `Mysql` |
```

Comment:
It should be: 
B is executed only, because that is the only test not marked with `Mysql`

Reply:
Fixed